ext/strings.go

```diff
@@ -485,6 +550,490 @@ func join(strs []string) (string, error) {
 	return strings.Join(strs, ""), nil
 }
 
+func stringFormat(s string, args traits.Lister) (string, error) {
+	return stringFormatWithLocale(s, args, defaultLocale)
+}
+
+func stringFormatWithLocale(formatStr string, argList traits.Lister, locale string) (string, error) {
+	i := 0
+	argIndex := 0
+	var builtStr strings.Builder
+	for i < len(formatStr) {
+		if formatStr[i] == '%' {
+			if i+1 < len(formatStr) && formatStr[i+1] == '%' {
+				err := builtStr.WriteByte('%')
+				if err != nil {
+					return "", fmt.Errorf("error writing format string: %w", err)
+				}
+				i += 2
+				continue
+			} else {
+				newStrIndex, val, refErr := parseFormatArg(i, argIndex, formatStr, argList, locale)
+				if refErr != nil {
+					return "", refErr
+				}
+				_, err := builtStr.WriteString(val)
+				if err != nil {
+					return "", fmt.Errorf("error writing format string: %w", err)
+				}
+				i = newStrIndex
+				argIndex++
+			}
+		} else {
+			err := builtStr.WriteByte(formatStr[i])
+			if err != nil {
+				return "", fmt.Errorf("error writing format string: %w", err)
+			}
+			i++
+		}
+	}
+	return builtStr.String(), nil
+}
+
+func parseFormatArg(lastStrIndex, argIndex int, formatStr string, args traits.Lister, locale string) (int, string, error) {
```

Comment:
Somehow document return values? (named args? godoc?)

Reply:
(marking this as resolved, added godoc)